### ISSUE TITLE
How many epochs are needed for training DETR with a custom dataset ?

### ISSUE THREAD

**Issue by stamatelou** [facebookresearch/detr#341]
I want to train DETR with a custom dataset with 14 classes. 
For at least how many epochs do I need to train it?

**Comment by lessw2020** [comment]
Hi @stamatelou  - 
To your question, it's really going to depend based on the specific dataset and how DETR ends up interacting with it. 
 As a general idea, I have trained with as few as 40, to as many as 120 epochs, depending on the complexity and the results during training.  My datasets have a lot of relationships within the objects though so it's favorable to DETR in general. 
If you watch your training curves you'll see when it peaks out, and when the validation results start to flatten out even when training results continue to drop.
That said, there's also a feedback loop in that if you add more /better augmentations to the same dataset, then you can find that longer training will continue to move those peaks further out, typically indicating model improvements and you should train longer to maximize that.
Note that I have shifted the learning rate drops around based on experience, so that I'm often dropping the rate at 30, 45, 60 and gamma of .3... which is different than the original training drops as I recall. 

That's some general feedback - to your specific question of "at least how many" I would always train with at least 30. 
Hope that helps!
